### ISSUE TITLE
Postgres timestamptz timezone disappears in resultset of PreparedStatement after 5 queries

### ISSUE THREAD

**Issue by TrustNoOne** [brettwooldridge/HikariCP#473]
Hi,
my slick mapping in play2.4 wasn't working, and I boiled down the problem to this: If I do a simple select from a table with a `timestamp with time zone` column, the timezone disappears from the results after a while. The example at the end of this message produces the following output: 
```
2015-10-27 20:45:13.459+01
2015-10-27 20:45:13.459+01
2015-10-27 20:45:13.459+01
2015-10-27 20:45:13.459+01
2015-10-27 20:45:13.459+01
// from now on, the timezone is never returned (even after 1000 queries)
2015-10-27 20:45:13.459
2015-10-27 20:45:13.459
2015-10-27 20:45:13.459
2015-10-27 20:45:13.459
...
```

- If I create the connection directly without using HikariCP, it works. 
- **If I don't close the connection at the end of every query (i.e. I don't release the old one), it works**
- If I use the same connection without creating/closing a new one (i.e. getting one from the pool and releasing it) each time, it works
- If I create and close a new connection each time using standard DriverManager.getConnection with the same URL, it works.
- **If I don't prepare the statement, it works**
- **If I prepare the statement twice (even without using the second one), it works**
- If I prepare a second statement different from the first one, it doesn't work

I tried both normal and java6 versions. 

You can reproduce the issue with a project with just the HikariCP and the postgres dependencies:

```java

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

public class TestHikari {
    public static void main(String[] args) throws Exception {
        HikariConfig conf = new HikariConfig();
        conf.setJdbcUrl("jdbc:postgresql://localhost/postgres");
        conf.setDriverClassName("org.postgresql.Driver");

        HikariDataSource ds = new HikariDataSource(conf);
        Connection ddlconn = ds.getConnection();
        try { ddlconn.createStatement().execute("DROP TABLE TEST"); } catch(Exception ignored) {}
        ddlconn.createStatement().execute("CREATE TABLE TEST (ts TIMESTAMP with time zone)");
        ddlconn.createStatement().execute("insert into test(ts) values('2015-10-27 20:45:13.459+01')");
        ddlconn.close();

        for (int i = 0; i< 10; i++) {
            Connection conn = ds.getConnection();
            PreparedStatement stmt = conn.prepareStatement("select ts from TEST");
            ResultSet rs = stmt.executeQuery();
            rs.next();

            System.out.println(rs.getString(1));

            conn.close();
        }
    }
}
```

**Comment by jnehlmeier** [comment]
Try a snapshot build of Postgres JDBC driver. 

There were some issues with dates / timezones when the JDBC driver switches to server-side prepared statements. By default this happens after 5 prepared statement executions (see prepareThreshold parameter of postgresql JDBC).

**Comment by brettwooldridge** [comment]
@TrustNoOne What driver version?

**Comment by brettwooldridge** [comment]
Closed as driver bug https://github.com/pgjdbc/pgjdbc/issues/130

**Comment by TrustNoOne** [comment]
@brettwooldridge tried all the available 9.4 jdbc4/41/42 (including snapshots)
That bug doesn't seem to go away. It's open since march 2014... Thanks for pointing me to the problem
:+1:

**Comment by brettwooldridge** [comment]
@TrustNoOne It seems the only workaround for now is *prepareThreshold=0*.  I recommend commenting on the driver issue in hopes of forcing some action to fix it.
